source/frontend/src/models/generate/GenerateAgreement.ts

```diff
@@ -0,0 +1,32 @@
+import { Api_Agreement } from 'models/api/Agreement';
+import moment from 'moment';
+import { formatMoney } from 'utils';
+
+import { Api_GenerateFile } from './GenerateFile';
+export class Api_GenerateAgreement {
+  file: Api_GenerateFile;
+  current_year: string;
+  date: string;
+  completion_date: string;
+  termination_date: string;
+  commencement_date: string;
+  status: string;
+  deposit_amount: string;
+  purchase_price: string;
+  survey_plan_number: string;
+  no_later_than_days: string;
+
+  constructor(agreement: Api_Agreement, file: Api_GenerateFile) {
```

Comment:
nit: file-> generateFile (imo file implies that this is the api_acqfile vs the generate kind)